extension/src/LanguageFunctions.ts

```diff
@@ -1005,18 +1005,27 @@ async function getBaseAppTranslationMap(
     await baseAppTranslationFiles.getBlobs([targetFilename]);
     localTransFiles = localBaseAppTranslationFiles();
   }
+
   const baseAppJsonPath = localTransFiles.get(targetFilename);
+  let parsedBaseApp = {};
   if (baseAppJsonPath !== undefined) {
     const baseAppJsonContent = readFileSync(baseAppJsonPath, "utf8");
     if (baseAppJsonContent.length === 0) {
       throw new Error(`No content in file: "${baseAppJsonPath}".`);
     }
-    const baseAppTranslationMap: Map<string, string[]> = new Map(
-      Object.entries(JSON.parse(baseAppJsonContent))
-    );
-    return baseAppTranslationMap;
+    try {
+      parsedBaseApp = JSON.parse(baseAppJsonContent);
+    } catch (err) {
+      throw new Error(
+        `Could not parse match file for "${targetFilename}". Message: ${
+          (err as Error).message
+        }. If this persists, try disabling the setting "NAB: Match Base App Translation". File path: ${baseAppJsonPath}`
```

Comment:
...and log an issue on https://github.com/jwikman/nab-al-tools/issues?
It would be nice to figure out how common this is and getting to the root cause one day...

Reply:
This could only happen if file was successfully downloaded, but corrupted later, right? We do already check parsing after download...

We should also delete the local file here, so it'll get downloaded in the next try.

Reply:
Yes that is correct. It shouldn't really be needed anymore. I think we should keep them for now, it's a safeguard that provides valuable information to the user in the, unlikely, case of this happening.